plugins/teststeps/awsfileupload/awsFileUpload.go

```diff
@@ -0,0 +1,299 @@
+	s3bucket        string      // AWS Bucket name
+	s3path          string      // Path where in the bucket to upload the file
+	s3credfile      string      // Path to the AWS credential file to authenticate the session
+	s3credprofile   string      // Profile of the AWS credential file that shall be used
+	emitURLtoReport bool        // Bool that defines that URLtoReport Event shall be triggered
+	compress        bool        // Bool that defines if the data should be compressed before uploading it
+}
+
+// Datatype for the emmiting the URLtoReport Event
+type eventURLtoReportPayload struct {
+	Msg string
+}
+
+// Name returns the plugin name.
+func (ts FileUpload) Name() string {
+	return Name
+}
+
+func emitEvent(ctx xcontext.Context, name event.Name, payload interface{}, tgt *target.Target, ev testevent.Emitter) error {
+	payloadStr, err := json.Marshal(payload)
+	if err != nil {
+		return fmt.Errorf("cannot encode payload for event '%s': %v", name, err)
+	}
+	rm := json.RawMessage(payloadStr)
+	evData := testevent.Data{
+		EventName: name,
+		Target:    tgt,
+		Payload:   &rm,
+	}
+	if err := ev.Emit(ctx, evData); err != nil {
+		return fmt.Errorf("cannot emit event EventURLtoReport: %v", err)
+	}
+	return nil
+}
+
+// Run executes the awsFileUpload.
+func (ts *FileUpload) Run(ctx xcontext.Context, ch test.TestStepChannels, params test.TestStepParameters,
+	ev testevent.Emitter, resumeState json.RawMessage) (json.RawMessage, error) {
+	// Validate the parameter
+	if err := ts.validateAndPopulate(params); err != nil {
+		return nil, err
+	}
+	f := func(ctx xcontext.Context, target *target.Target) error {
+		// expand args
+		path, err := ts.path.Expand(target)
+		if err != nil {
+			return fmt.Errorf("failed to expand argument '%s': %v", ts.path, err)
+		}
+		filename, err := ts.filename.Expand(target)
+		if err != nil {
+			return fmt.Errorf("failed to expand argument dir '%s': %v", ts.filename, err)
+		}
+		var url string
+		// Compress if compress parameter is true
+		if ts.compress {
+			// Create buffer for the compressed data
+			var buf bytes.Buffer
+			// Create the archive and write the output to the "out" Writer
+			buf, err = createTarArchive(path, buf)
+			if err != nil {
+				return fmt.Errorf("error creating an archive: %v", err)
+			}
+			fmt.Println("Tar archive created successfully")
+			filename = filename + ".tar.gz"
+			// Upload file
+			url, err = ts.upload(filename, buf.Bytes())
+			if err != nil {
+				return fmt.Errorf("could not upload the file: %v", err)
+			}
+		} else {
+			// Read the file that should be uploaded
+			bodyBytes, err := ioutil.ReadFile(path)
+			if err != nil {
+				return fmt.Errorf("could not read the file: %v", err)
+			}
+			// Upload file
+			url, err = ts.upload(filename, bodyBytes)
+			if err != nil {
+				return fmt.Errorf("could not upload the file: %v", err)
+			}
+		}
+		// Emit URLtoReport event to get the url into the report
+		if ts.emitURLtoReport {
```

Comment:
I think this should be done every time, without gating it behind a parameter. What would be a reason not to emit this event?